diffsims/generators/simulation_generator.py

```diff
@@ -0,0 +1,370 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2023 The diffsims developers
+#
+# This file is part of diffsims.
+#
+# diffsims is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# diffsims is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Union, Sequence
+import numpy as np
+
+from orix.quaternion import Rotation
+from orix.crystal_map import Phase
+
+from diffsims.crystallography import ReciprocalLatticeVector
+from diffsims.utils.shape_factor_models import (
+    linear,
+    atanc,
+    lorentzian,
+    sinc,
+    sin2c,
+    lorentzian_precession,
+    _shape_factor_precession,
+)
+
+from diffsims.utils.sim_utils import (
+    get_electron_wavelength,
+    get_kinematical_intensities,
+    is_lattice_hexagonal,
+    get_points_in_sphere,
+    get_intensities_params,
+)
+
+_shape_factor_model_mapping = {
+    "linear": linear,
+    "atanc": atanc,
+    "sinc": sinc,
+    "sin2c": sin2c,
+    "lorentzian": lorentzian,
+}
+
+from diffsims.simulations import Simulation1D, Simulation2D
+
+
+class SimulationGenerator:
+    """
+    A class for generating kinematic diffraction simulations.
+    """
+
+    def __init__(
+        self,
+        accelerating_voltage: float = 200,
+        scattering_params: str = "lobato",
+        precession_angle: float = 0,
+        shape_factor_model: str = "lorentzian",
+        approximate_precession: bool = True,
+        minimum_intensity: float = 1e-20,
+        **kwargs,
+    ):
```

Comment:
This should be documented. Although some units will be clear (`accelerating_voltage`) others will probably not be.